frontend/src/components/TransactionsList.tsx

```diff
@@ -13,15 +25,42 @@ function TransactionsList({
   appId,
   showReceiveButton = true,
 }: TransactionsListProps) {
-  const { data: transactions, isLoading } = useTransactions(appId);
+  const [page, setPage] = useState(1);
+  const { data: transactionData, isLoading } = useTransactions(
+    appId,
+    false,
+    LIST_TRANSACTIONS_LIMIT,
+    page
+  );
+  const transactions = transactionData?.transactions || [];
+  const totalCount = transactionData?.totalCount || 0;
+  const totalPages = Math.ceil(totalCount / LIST_TRANSACTIONS_LIMIT);
+
+  const pageNumbers = useMemo(() => {
+    return generatePageNumbers(page, totalPages);
+  }, [page, totalPages]);
 
-  if (isLoading) {
+  useEffect(() => {
+    const el = document.querySelector(".transaction-list");
+    if (el) {
+      el.scrollIntoView({ behavior: "smooth", block: "start" });
```

Comment:
why is this needed?

Reply:
So we can scroll to top on changing the page

Reply:
minor: there should be a react way to do this using a ref, right?

Reply:
Done!